### PR TITLE
LibraryExporter : added CompletionStatus column export

### PR DESCRIPTION
**Pull request by luskynavy**
Hi,

It just adds the info Completion Status to the export as a column.
Parametrable like others column in settings.

image
image

Thanks by advance.

I have verified that:
- [x] These changes work, by building the extension and testing.
- [x] That the changes comply with the [rules](https://github.com/darklinkpower/PlayniteExtensionsCollection#contributing) indicated in the repository.
- [x] Pull request is targeting `master` branch.

### CODE REVIEW

File: source/Generic/LibraryExporter/Views/LibraryExporterView.xaml [hunk 1]
```diff
@@ -64,6 +64,8 @@
                       Content="{DynamicResource LOCRecentActivityLabel}" Margin="0,0,0,5" />
                 <CheckBox IsChecked="{Binding Settings.Settings.ExportSettings.Roms}"
                       Content="Roms" Margin="0,0,0,5" />
+                <CheckBox IsChecked="{Binding Settings.Settings.ExportSettings.CompletionStatus}"
+                      Content="CompletionStatus" Margin="0,0,0,5" />
```
Comment (darklinkpower) [inline source/Generic/LibraryExporter/Views/LibraryExporterView.xaml:68]: The Content should use the Playnite localization resource like the other properties

File: source/Generic/LibraryExporter/ViewModels/LibraryExporterViewModel.cs [hunk 2]
```diff
@@ -244,6 +245,12 @@ private string[] GenerateGameColumns(int columnCount, Game game)
                 currentInsertColumn++;
             }
 
+            if (Settings.Settings.ExportSettings.CompletionStatus)
+            {
+                properties[currentInsertColumn] = !string.IsNullOrEmpty(game.CompletionStatus.Name) ? game.CompletionStatus.Name : string.Empty;
```
Comment (darklinkpower) [inline source/Generic/LibraryExporter/ViewModels/LibraryExporterViewModel.cs:250]: I think here it should be checked if `CompletionStatus` is null. Also for consistency use the already available `IsNullOrEmpty` extension method instead, like `!game.CompletionStatus.Name.IsNullOrEmpty()`

File: source/Generic/LibraryExporter/ViewModels/LibraryExporterViewModel.cs [hunk 3]
```diff
@@ -99,6 +99,7 @@ private List<string> GetColumnsList()
             if (Settings.Settings.ExportSettings.Modified) { columnNames.Add(GetLocalizationString("LOCDateModifiedLabel")); }
             if (Settings.Settings.ExportSettings.RecentActivity) { columnNames.Add(GetLocalizationString("LOCRecentActivityLabel")); }
             if (Settings.Settings.ExportSettings.Roms) { columnNames.Add("Roms"); }
+            if (Settings.Settings.ExportSettings.CompletionStatus) { columnNames.Add("CompletionStatus"); }
```
Comment (darklinkpower) [inline source/Generic/LibraryExporter/ViewModels/LibraryExporterViewModel.cs:102]: Should also use Playnite localization string

File: source/Generic/LibraryExporter/Views/LibraryExporterView.xaml (hunk 1 above)
Comment (luskynavy) [reply source/Generic/LibraryExporter/Views/LibraryExporterView.xaml:68]: Ok I will use LOCCompletionStatus.

File: source/Generic/LibraryExporter/ViewModels/LibraryExporterViewModel.cs (hunk 3 above)
Comment (luskynavy) [reply source/Generic/LibraryExporter/ViewModels/LibraryExporterViewModel.cs:102]: Same here I will use LOCCompletionStatus.

File: source/Generic/LibraryExporter/ViewModels/LibraryExporterViewModel.cs (hunk 2 above)
Comment (luskynavy) [reply source/Generic/LibraryExporter/ViewModels/LibraryExporterViewModel.cs:250]: Ok, I guess it's ok like that:
`properties[currentInsertColumn] = (game.CompletionStatus != null && !game.CompletionStatus.Name.IsNullOrEmpty() ? game.CompletionStatus.Name : string.Empty);`

**Comment by luskynavy** [comment]
I made changes here [LibraryExporter : localization and checks for completion status export](https://github.com/darklinkpower/PlayniteExtensionsCollection/pull/417/commits/df2b62edcf00ea4d71ec98f6b769f4b878d76948) and re-requested a review (1st time using github PR, not sure it's the normal method).

**Comment by darklinkpower** [comment]
> I made changes here [LibraryExporter : localization and checks for completion status export](https://github.com/darklinkpower/PlayniteExtensionsCollection/pull/417/commits/df2b62edcf00ea4d71ec98f6b769f4b878d76948) and re-requested a review (1st time using github PR, not sure it's the normal method).

You need to mark the requested changes conversations as resolved by clicking here.

image

After that you can either request or not a review but not always necessary.

**Pull request merged**